k2/csrc/log.h

```diff
@@ -178,11 +197,12 @@ class Voidifier {
 #define K2_CHECK_CUDA_ERROR(x) \
   K2_CHECK_EQ(x, cudaSuccess) << " Error: " << cudaGetErrorString(x) << ". "
 
-#define K2_CUDA_SAFE_CALL(...)               \
-  do {                                       \
-    (__VA_ARGS__);                           \
-    cudaDeviceSynchronize();                 \
-    K2_CHECK_CUDA_ERROR(cudaGetLastError()); \
+#define K2_CUDA_SAFE_CALL(...)                                             \
+  do {                                                                     \
+    (__VA_ARGS__);                                                         \
+    ::k2::internal::kDisableDebug ? cudaSuccess : cudaDeviceSynchronize(); \
```

Comment:
why is  `cudaSuccess` here?  Do you mean some code like `void (0)`?

Reply:
I would suggest
```cpp
::k2::internal::kDisableDebug ? (void)0 : (void)cudaDeviceSynchronize();
```

since we do not care about the result.

Reply:
changed it to a new sensible way. As `(void)0` give error: `operand types are incompatible ("void" and "cudaError_t")`. And if the return of cudaDeviceSynchronize gets checked, check cudaGetLastError again is excess.